src/main/java/uk/gov/companieshouse/chs/gov/uk/notify/integration/api/service/GovUkNotifyService.java

```diff
@@ -72,13 +69,13 @@ public LetterResp sendLetter(
             return new LetterResp(response != null && response.getNotificationId() != null,
                     response);
         } catch (NotificationClientException nce) {
-            var logData = Map.of("reference", reference);
-            LOGGER.error("Failed to send letter", nce, new HashMap<>(logData));
+            Map<String, Object> logData = createLogData(reference);
+            LOGGER.error("Failed to send letter", nce, logData);
             try {
                 var response = buildLetterResponseForError(nce, reference);
```

Comment:
Do we need "response" variable ? We could combine line 75 and 76 for a concise code.